### ISSUE TITLE
"ERROR    invalid license token, try to run `pyarmor reg` to register license again", while executing on a Linux VM

### ISSUE THREAD

**Issue by goraje** [dashingsoft/pyarmor#1551]
```
INFO     Python 3.10.13
INFO     Pyarmor 8.4.1 (group), 006005, ******
INFO     Platform linux.aarch64
```

Hi, after issue [#1542](https://github.com/dashingsoft/pyarmor/issues/1542) was resolved I've started working with Pyarmor on Docker installed on a Linux VM (Ubuntu 22.04 aarch64) and Pyarmor installed on the Linux VM itself. Independently of the environment that I've used I have been getting ``ERROR    invalid license token, try to run `pyarmor reg` to register license again`` while trying to obfuscate a Python package in a single run.

The command that I used was:
```
pyarmor --debug gen --obf-code 2 --enable-jit -O armored -e .2023-11-01 -i src/******
``` 

The command does not outright fail but rather an error is encountered after successfully obfuscating a couple of modules.

The output looks as follows:
```
INFO     Python 3.10.13
INFO     Pyarmor 8.4.1 (group), 006005, ******
INFO     Platform linux.aarch64
DEBUG    native platform linux.aarch64
DEBUG    home path: /home/jgora/.pyarmor
DEBUG    command options: {'output': 'armored', 'no_runtime': False, 'enable_jit': True, 'obf_code': 2, 'import_prefix': 1, 'expired': '.2023-11-01', 'inputs': ['src/******']}
DEBUG    install plugin: CodesignPlugin
DEBUG    install plugin: DarwinUniversalPlugin
INFO     search inputs ...
INFO     find package at src/******
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'linux.aarch64'}
DEBUG    got /home/jgora/pyarmor-venv/lib/python3.10/site-packages/pyarmor/cli/core/pyarmor_runtime.so
INFO     write armored/******/pyarmor_runtime_006005/pyarmor_runtime.so
INFO     patch runtime file
DEBUG    patching runtime data at 579680
DEBUG    patch runtime file OK
DEBUG    call post runtime plugin <class 'pyarmor.cli.plugin.CodesignPlugin'>
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "******"
INFO     obfuscating file dataclass_utils.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/dataclass_utils.py
INFO     obfuscating file defaults.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/defaults.py
INFO     obfuscating file version.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/version.py
INFO     obfuscating file sockutil.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RAISE_VARARGS" at <frozen ******.sockutil>:23:safe_localhost
DEBUG    special co "RAISE_VARARGS" at <frozen ******.sockutil>:92:_validate_addresses
INFO     write armored/******/sockutil.py
INFO     obfuscating file paths.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/paths.py
INFO     obfuscating file types.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/types.py
INFO     obfuscating file domains.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RAISE_VARARGS" at <frozen ******.domains>:462:__getattr__
INFO     write armored/******/domains.py
INFO     obfuscating file logging_context.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/logging_context.py
INFO     obfuscating file settings.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/settings.py
INFO     obfuscating file uploaders.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RAISE_VARARGS" at <frozen ******.uploaders>:86:upload
DEBUG    special co "RAISE_VARARGS" at <frozen ******.uploaders>:204:get_uploader
INFO     write armored/******/uploaders.py
INFO     obfuscating file constants.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/constants.py
INFO     obfuscating file exception_helpers.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/exception_helpers.py
INFO     obfuscating file resource_monitor.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/resource_monitor.py
INFO     obfuscating file truststore.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RERAISE" at <frozen ******.truststore>:626:certificate_chain
INFO     write armored/******/truststore.py
INFO     obfuscating file jwt.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RERAISE" at <frozen ******.access.jwt>:42:decode_jwt
INFO     write armored/******/access/jwt.py
INFO     obfuscating file __init__.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/access/__init__.py
INFO     obfuscating file claims.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RERAISE" at <frozen ******.access.claims>:748:validate_email
INFO     write armored/******/access/claims.py
INFO     obfuscating file oauth.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/access/oauth.py
INFO     obfuscating file extensions.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RAISE_VARARGS" at <frozen ******.access.extensions>:45:value
DEBUG    special co "RAISE_VARARGS" at <frozen ******.access.extensions>:52:from_public_bytes
INFO     write armored/******/access/extensions.py
INFO     obfuscating file exceptions.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/access/exceptions.py
INFO     obfuscating file ca.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RERAISE" at <frozen ******.access.ca>:103:validate_email
DEBUG    special co "RERAISE" at <frozen ******.access.ca>:118:validate_email
DEBUG    special co "RERAISE" at <frozen ******.access.ca>:361:get_certificate_extension_value
DEBUG    special co "RAISE_VARARGS" at <frozen ******.access.ca>:380:_parse_certificate_subject_name
INFO     write armored/******/access/ca.py
INFO     obfuscating file users.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RERAISE" at <frozen ******.access.users>:76:validate_email
DEBUG    special co "RERAISE" at <frozen ******.access.users>:185:validate_email
INFO     write armored/******/access/users.py
INFO     obfuscating file generators.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/access/generators.py
INFO     obfuscating file token.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    unknown error, please check pyarmor.error.log
ERROR    invalid license token, try to run `pyarmor reg` to register license again
```

Other things that might be of importance:
- if I redo the registration process and rerun the command it stops at exactly the same point of execution
- the same behavior can be observed if I run it inside of a Docker container on this Linux host
- I've verified that the output of `Pytransform3.get_hd_info(22)` remains constant for this VM between reboots
- I can successfully obfuscate this package using the same setup and command from my MacBook laptop

**Comment by jondy** [comment]
> I've verified that the output of  Pytransform3.get_hd_info(22) remains constant for this VM between reboots 

Try to generate new device file by Pyarmor 8.4.1 for this VM

**Comment by jondy** [comment]
And what's the output of `pyarmor -d reg xxxx` and `pyarmor -v` in this VM?

**Comment by goraje** [comment]
Hi, firstly I did a clean rerun with the "old" `device-regfile`:
- I verified the installed version of Pyarmor
```
$ pip list | grep pyarmor
pyarmor          8.4.1
pyarmor.cli.core 5.4.0
```
- I backed-up `~/.pyarmor`
```
$ mv ~/.pyarmor ~/.pyarmor_bkp
```
- I ran `pyarmor -d reg`
```
$ pyarmor -d reg pyarmor-device-regfile-6005.11.zip
INFO     Python 3.10.13
INFO     Pyarmor 8.4.1 (trial), 000000, non-profits
INFO     Platform linux.aarch64
DEBUG    native platform linux.aarch64
DEBUG    home path: /home/jgora/.pyarmor
INFO     register "pyarmor-device-regfile-6005.11.zip"
DEBUG    extracting license.lic
DEBUG    extracting .pyarmor_capsule.zip
INFO     machine id in group license: m88f687b18066063d0b2c1ba3a844ca1c
INFO     got machine id: m88f687b18066063d0b2c1ba3a844ca1c
INFO     this machine id matchs group license
DEBUG    extracting tokens/m88f687b18066063d0b2c1ba3a844ca1c
INFO     This license registration information:

License Type    : pyarmor-group
License No.     : pyarmor-vax-006005
License To      : *****
License Product : ******

BCC Mode        : Yes
RFT Mode        : Yes

Notes
* Offline obfuscation
```
- I executed `pyarmor -v`
```
$ pyarmor -v
Pyarmor 8.4.1 (group), 006005, ******

License Type    : pyarmor-group
License No.     : pyarmor-vax-006005
License To      : *****
License Product : ******

BCC Mode        : Yes
RFT Mode        : Yes

Notes
* Offline obfuscation
```
- I tried to obfuscated the package again (and it failed in the same way as previously so I redacted the middle part of the log)
```
$ pyarmor --debug gen --obf-code 2 -O armored -e .2023-11-01 -i src/******
INFO     Python 3.10.13
INFO     Pyarmor 8.4.1 (group), 006005, ******
INFO     Platform linux.aarch64
DEBUG    native platform linux.aarch64
DEBUG    home path: /home/jgora/.pyarmor
DEBUG    command options: {'output': 'armored', 'no_runtime': False, 'obf_code': 2, 'import_prefix': 1, 'expired': '.2023-11-01', 'inputs': ['src/******']}
DEBUG    install plugin: CodesignPlugin
DEBUG    install plugin: DarwinUniversalPlugin
INFO     search inputs ...
INFO     find package at src/******
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'linux.aarch64'}
DEBUG    got /home/jgora/pyarmor-venv/lib/python3.10/site-packages/pyarmor/cli/core/pyarmor_runtime.so
INFO     write armored/******/pyarmor_runtime_006005/pyarmor_runtime.so
INFO     patch runtime file
DEBUG    patching runtime data at 579680
DEBUG    patch runtime file OK
DEBUG    call post runtime plugin <class 'pyarmor.cli.plugin.CodesignPlugin'>
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "******"
INFO     obfuscating file dataclass_utils.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/dataclass_utils.py
...
INFO     obfuscating file exceptions.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/access/exceptions.py
INFO     obfuscating file ca.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    unknown error, please check pyarmor.error.log
ERROR    invalid license token, try to run `pyarmor reg` to register license again
```
- I ran `pyarmor -v` again after the failed execution
```
$ pyarmor -v
Pyarmor 8.4.1 (group), 006005, ******

License Type    : pyarmor-group
License No.     : pyarmor-vax-006005
License To      : *****
License Product : ******

BCC Mode        : Yes
RFT Mode        : Yes

Notes
* Offline obfuscation
```
Then I did a new registration procedure.
```
$ mv ~/.pyarmor ~/.pyarmor_bkp

$ pyarmor reg -g 12
INFO     Python 3.10.13
INFO     Pyarmor 8.4.1 (trial), 000000, non-profits
INFO     Platform linux.aarch64
INFO     generating device file ".pyarmor/group/pyarmor-group-device.12"
INFO     current machine id is "m88f687b18066063d0b2c1ba3a844ca1c"
INFO     device file has been generated successfully

$ pyarmor reg -g 12 pyarmor-regfile-6005.zip
INFO     Python 3.10.13
INFO     Pyarmor 8.4.1 (trial), 000000, non-profits
INFO     Platform linux.aarch64
INFO     register device file ".pyarmor/group/pyarmor-group-device.12"
INFO     use group license "pyarmor-regfile-6005.zip"
INFO     send request to server
INFO     write registration file "pyarmor-device-regfile-6005.12.zip"
INFO     please copy deivce regfile to offline device and run
INFO         pyarmor reg pyarmor-device-regfile-6005.12.zip
INFO     The device regfile has been generated successfully

$ pyarmor -d reg pyarmor-device-regfile-6005.12.zip
INFO     Python 3.10.13
INFO     Pyarmor 8.4.1 (trial), 000000, non-profits
INFO     Platform linux.aarch64
DEBUG    native platform linux.aarch64
DEBUG    home path: /home/jgora/.pyarmor
INFO     register "pyarmor-device-regfile-6005.12.zip"
DEBUG    extracting license.lic
DEBUG    extracting .pyarmor_capsule.zip
INFO     machine id in group license: m88f687b18066063d0b2c1ba3a844ca1c
INFO     got machine id: m88f687b18066063d0b2c1ba3a844ca1c
INFO     this machine id matchs group license
DEBUG    extracting tokens/m88f687b18066063d0b2c1ba3a844ca1c
INFO     This license registration information:

License Type    : pyarmor-group
License No.     : pyarmor-vax-006005
License To      : *****
License Product : ******

BCC Mode        : Yes
RFT Mode        : Yes

Notes
* Offline obfuscation

$ pyarmor -v
Pyarmor 8.4.1 (group), 006005, ******

License Type    : pyarmor-group
License No.     : pyarmor-vax-006005
License To      : *****
License Product : ******

BCC Mode        : Yes
RFT Mode        : Yes

Notes
* Offline obfuscation
```
After the new registration I ran the obfuscation, but once again it failed at exactly the same spot as before.
```
$ pyarmor --debug gen --obf-code 2 -O armored -e .2023-11-01 -i src/******
INFO     Python 3.10.13
INFO     Pyarmor 8.4.1 (group), 006005, ******
INFO     Platform linux.aarch64
DEBUG    native platform linux.aarch64
DEBUG    home path: /home/jgora/.pyarmor
DEBUG    command options: {'output': 'armored', 'no_runtime': False, 'obf_code': 2, 'import_prefix': 1, 'expired': '.2023-11-01', 'inputs': ['src/******']}
DEBUG    install plugin: CodesignPlugin
DEBUG    install plugin: DarwinUniversalPlugin
INFO     search inputs ...
INFO     find package at src/******
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'linux.aarch64'}
DEBUG    got /home/jgora/pyarmor-venv/lib/python3.10/site-packages/pyarmor/cli/core/pyarmor_runtime.so
INFO     write armored/******/pyarmor_runtime_006005/pyarmor_runtime.so
INFO     patch runtime file
DEBUG    patching runtime data at 579680
DEBUG    patch runtime file OK
DEBUG    call post runtime plugin <class 'pyarmor.cli.plugin.CodesignPlugin'>
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "******"
INFO     obfuscating file dataclass_utils.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/dataclass_utils.py
INFO     obfuscating file licensing.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RERAISE" at <frozen ******.licensing>:101:get_banner
DEBUG    special co "RERAISE" at <frozen ******.licensing>:111:get_code
DEBUG    special co "RERAISE" at <frozen ******.licensing>:118:get_days_left
DEBUG    special co "RERAISE" at <frozen ******.licensing>:126:get_expiration
INFO     write armored/******/licensing.py
INFO     obfuscating file defaults.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/defaults.py
INFO     obfuscating file version.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/version.py
INFO     obfuscating file sockutil.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RAISE_VARARGS" at <frozen ******.sockutil>:23:safe_localhost
DEBUG    special co "RAISE_VARARGS" at <frozen ******.sockutil>:92:_validate_addresses
INFO     write armored/******/sockutil.py
INFO     obfuscating file paths.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/paths.py
INFO     obfuscating file types.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/types.py
INFO     obfuscating file domains.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RAISE_VARARGS" at <frozen ******.domains>:462:__getattr__
INFO     write armored/******/domains.py
INFO     obfuscating file logging_context.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/logging_context.py
INFO     obfuscating file settings.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/settings.py
INFO     obfuscating file uploaders.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RAISE_VARARGS" at <frozen ******.uploaders>:86:upload
DEBUG    special co "RAISE_VARARGS" at <frozen ******.uploaders>:204:get_uploader
INFO     write armored/******/uploaders.py
INFO     obfuscating file constants.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/constants.py
INFO     obfuscating file exception_helpers.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/exception_helpers.py
INFO     obfuscating file resource_monitor.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/resource_monitor.py
INFO     obfuscating file truststore.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RERAISE" at <frozen ******.truststore>:626:certificate_chain
INFO     write armored/******/truststore.py
INFO     obfuscating file jwt.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RERAISE" at <frozen ******.access.jwt>:42:decode_jwt
INFO     write armored/******/access/jwt.py
INFO     obfuscating file __init__.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/access/__init__.py
INFO     obfuscating file claims.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RERAISE" at <frozen ******.access.claims>:748:validate_email
INFO     write armored/******/access/claims.py
INFO     obfuscating file oauth.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/access/oauth.py
INFO     obfuscating file extensions.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
DEBUG    special co "RAISE_VARARGS" at <frozen ******.access.extensions>:45:value
DEBUG    special co "RAISE_VARARGS" at <frozen ******.access.extensions>:52:from_public_bytes
INFO     write armored/******/access/extensions.py
INFO     obfuscating file exceptions.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     write armored/******/access/exceptions.py
INFO     obfuscating file ca.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    process attribute
DEBUG    unknown error, please check pyarmor.error.log
ERROR    invalid license token, try to run `pyarmor reg` to register license again
```
After the last execution I also ran:
```
$ pyarmor -v
Pyarmor 8.4.1 (group), 006005, ******

License Type    : pyarmor-group
License No.     : pyarmor-vax-006005
License To      : *****
License Product : ******

BCC Mode        : Yes
RFT Mode        : Yes

Notes
* Offline obfuscation
```

**Comment by Sakura4036** [comment]
Same Error on Linux_x86

```
INFO     Python 3.9.18
INFO     Pyarmor 8.4.1 (group), 006112, xxxxx
INFO     Platform linux.x86_64

pyarmor gen -O Pack-Linux_X86_64  -r Beta-Pack  --enable-jit --obf-code 2 --outer --mix-str  --no-wrap  --enable-bcc  --platform linux.x86_64
```

Error Info:
```
...
INFO     start to generate bcc code ...
INFO     generate bcc code successfully
ERROR    invalid license token, try to run `pyarmor reg` to register license again
```

**Comment by jondy** [comment]
Reproduced, fixing it.

**Comment by jondy** [comment]
It will be fixed in next release 8.4.2

Verify it by downloading `pytransform3.so` to overwrite `/path/to/pyarmor/cli/core/pytransform3.so`

http://pyarmor.dashingsoft.com/downloads/temp/5.4.1/linux.aarch64/libs/cp3.10/pytransform3.so
http://pyarmor.dashingsoft.com/downloads/temp/5.4.1/linux.x86_64/libs/cp3.9/pytransform3.so

**Comment by goraje** [comment]
Hi, I've downloaded an replaced the `pytransform3.so` extension module and it fixed the problems that I've been having with both the Linux VM, as well as Docker on the aforementioned VM. I've noticed one more thing that I wanted to ask about, because I'm not sure it's intended behavior and I have no idea if it is related to the changes that you've made or not. When I execute `pyarmor --debug gen --obf-code 2 -O armored -e .2023-11-01 -i src/******` on the VM I get the following block of log lines only once:
```
INFO     patch runtime file
DEBUG    patching runtime data at 579680
DEBUG    patch runtime file OK
```
However if I execute the same command inside of Docker container it gets repeated a few times during the start of the execution:
```
DEBUG    command options: {'output': 'armored', 'no_runtime': False, 'obf_code': 2, 'import_prefix': 1, 'expired': '.2023-11-01', 'inputs': ['src/******']}
DEBUG    install plugin: CodesignPlugin
DEBUG    install plugin: DarwinUniversalPlugin
INFO     search inputs ...
INFO     find package at src/******
INFO     find 1 top resources
INFO     patch runtime file
DEBUG    patching runtime data at 579680
DEBUG    patch runtime file OK
INFO     patch runtime file
DEBUG    patching runtime data at 579680
DEBUG    patch runtime file OK
INFO     patch runtime file
DEBUG    patching runtime data at 579680
DEBUG    patch runtime file OK
INFO     start to generate runtime files
INFO     target platforms {'linux.aarch64'}
```
and then a few times per each obfuscated module (which does not happen while running on the VM at all) eg.
```
INFO     obfuscating file dataclass_utils.py
DEBUG    process inline marker
DEBUG    parse script
INFO     patch runtime file
DEBUG    patching runtime data at 579680
DEBUG    patch runtime file OK
DEBUG    process co
DEBUG    process attribute
DEBUG    patch co
INFO     patch runtime file
DEBUG    patching runtime data at 579680
DEBUG    patch runtime file OK
INFO     write armored/******/dataclass_utils.py
```

**Comment by jondy** [comment]
You're terrific guy, :)

It's intended behavior, but it's sort of redundant. Maybe these logs will be removed in future.

**Comment by goraje** [comment]
In that case all that's left to say is thank you :) we really appreciate your work and swiftness with dealing with our requests.

**Comment by jondy** [comment]
Fixed in 8.4.2, note that also need upgrade pyarmor.cli.core by `pip install -U pyarmor.cli.core`
